### DISCUSSION TITLE
Product Option images or Product Variant images

### DISCUSSION THREAD

**Discussion by joekendal** [medusajs/medusa#9434] [Feature Requests]
Currently we can only put images at the Product level. But normally an e-commerce site would change the images based on the selected variant.

There has previously been a community maintained plugin but instead of migrating it to v2, I think it would be better to open a PR that implements this natively given the widespread adoption of this e-commerce functionality. 

Given that this would have an impact on the schema design, we are seeking guidance from the official medusa team.

**Comment by Deroswent** [comment]
the main discussion about this issue is here https://github.com/medusajs/medusa/discussions/4456

**Reply by Betanoir** [reply]
I think the only way to get this implemented is by raising multiple issues regarding the subject. The Medusa team either doesn't want to implement it for some reason or just doesn't see the suggestions.

**Reply by joekendal** [reply]
> I think the only way to get this implemented is by raising multiple issues regarding the subject. The Medusa team either doesn't want to implement it for some reason or just doesn't see the suggestions.

I spoke with one of the co-founders about it and he explained their reasons for why, after discussing it internally, they decided it was not a priority.

**Reply by Ksztof** [reply]
If such a crucial functionality isn't a priority for them, I wonder what their internal situation must be like.

**Reply by Betanoir** [reply]
Well it isn't a huge issue any more... I created a plugin to assist us with it - medusa-variant-images

**Reply by Ksztof** [reply]
It's working! Thanks a lot for your effort.

**Reply by Betanoir** [reply]
No problem! It doesn't work in production currently, but Medusa is releasing an update to fix it.

**Reply by Ksztof** [reply]
Do you know why it’s not working, or what exactly is the issue? I'm planning to host the store on my own server in about two months, if it works locally for me, it should work in production too (I hope), since it’s the same code.

**Reply by Betanoir** [reply]
It won't work in production as far as I'm aware. Medusa have posted something with regards to why it isn't working but they're publishing the fix in their next release, at which time I'll push the next update to the plugin that resolves the issue

**Reply by Ksztof** [reply]
Ok, thanks, hope they will fix it fast
